python/Makefile

```diff
@@ -69,9 +69,16 @@ $(EGG_FILE) : $(CRYPTO_SWIG_TARGET) $(KV_SWIG_TARGET) $(addprefix $(PROTOBUF_DPA
 	@ . $(abspath $(DSTDIR)/bin/activate) && \
 		BUILD_CLIENT=$(BUILD_CLIENT) python3 setup.py bdist_egg
 
-$(CRYPTO_SWIG_TARGET) $(KV_SWIG_TARGET) : $(CRYPTO_SWIG_SOURCE) $(KV_SWIG_SOURCE) $(COMMON_LIB) $(CRYPTO_LIB) $(KEYVAL_LIB)
+# In the following, we use the pattern rule with multiple targets
+# so that make knows that they are created through a single invocation of the target-body
+# Without that, when make runs with multiple threads, the target-body is executed
+# multiple times, possibly inducing build corruptions
+# This pattern rule can be avoided by using grouped targets starting in make 4.3
+pdo/common/crypto%py pdo/common/key_value_swig/key_value_swig%py : $(CRYPTO_SWIG_SOURCE) $(KV_SWIG_SOURCE) $(COMMON_LIB) $(CRYPTO_LIB) $(KEYVAL_LIB)
 	@ . $(abspath $(DSTDIR)/bin/activate) && \
 		BUILD_CLIENT=$(BUILD_CLIENT) python3 setup.py build_ext --force
+	test -e pdo/common/crypto$*py
+	test -e pdo/common/key_value_swig/key_value_swig$*py
 
```

Comment:
great catch! what is the value of the test commands at the end? and... would it make sense to separate these? i don't recall if build_ext allows specification of a module to build but that would make this process clearer and less buggy.

Reply:
> what is the value of the test commands at the end?

It's just to use the pattern rule with the `$*` -- I should check if applying it at the targets is sufficient.

> would it make sense to separate these?

That would avoid the grouping issue. But, as you say, I'm not sure that that's possible.

Reply:
Doing some research on this... my suggestion is that we take your proposal. When @marcelamelara proposes the full on conversion of our python build process, we should completely replace the make/setup with cmake and the package support that it provides for swig and python.